### ISSUE TITLE
Shipment Create via API salesShipmentRepositoryV1 throw Fatal error in Admin Order -> Shipment -> View

### ISSUE THREAD

**Issue by milindsingh** [magento/magento2#22686]
Shipment Create via API salesShipmentRepositoryV1 throw Fatal error in Admin Order -> Shipment -> View

### Preconditions (*)

1. Magento 2.3.1/2.3.0
2. The order created via Admin Panel

### Steps to reproduce (*)

1. Create an order created from Admin Panel
2. Create Shipment via API:

```json
{
  "entity": {
    "increment_id": "000000003",
    "order_id": 3,
    "items": [
        {
          "sku": "WSH12-32-Red",
          "qty": 1
        }
      ],
    "tracks": [
      {
        "order_id": 3,
        "qty": 1,
        "track_number": "123",
        "title": "Test",
        "carrier_code": "dhl"
      }
    ]  
  }
}
```
3. Goto `Admin -> Sales -> Order -> View -> Shipments -> View`

### Expected result (*)

1. Shipment Details should be rendered without any Fatal Error as API shipment creation was successful.

### Actual result (*)

1. Fatal Error in Admin Shipment View
```json
Fatal error: Uncaught Error: Call to a member function getParentItem() on null in /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/module-shipping/view/adminhtml/templates/view/items.phtml:19 Stack trace: #0 /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/framework/View/TemplateEngine/Php.php(59): include() #1 /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/framework/View/Element/Template.php(271): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Shipping\Block\Adminhtml\View\Items), '/opt/lampp7.2/h...', Array) #2 /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/framework/View/Element/Template.php(301): Magento\Framework\View\Element\Template->fetchView('/opt/lampp7.2/h...') #3 /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/module-backend/Block/Template.php(129): Magento\Framework\View\Element\Template->_toHtml() #4 /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/framework/View/Element/AbstractBlock.php(668): Magento\Backend\Block\Template->_toHtml() #5 /opt/lampp7 in /opt/lampp7.2/htdocs/magento/2.3.sample/vendor/magento/module-shipping/view/adminhtml/templates/view/items.phtml on line 19
```
![https i 17828 2 3 develop instances magento community engineering admin admin order_shipment view shipment_id 1](https://user-images.githubusercontent.com/13949703/50507095-fbade200-0aa1-11e9-89f0-3b26558f310d.png)

![screenshot from 2018-12-28 13-19-42](https://user-images.githubusercontent.com/13949703/50507392-5bf15380-0aa3-11e9-903a-94b8c3177d99.png)

**Related Issues:**
https://github.com/magento-engcom/msi/issues/2220
https://github.com/magento/magento2/issues/17828

**Note:**
I think the Shipment creation via API have missed some required data, which is not validated at time of processing the call, causing fatal errors.

**Comment by milindsingh** [comment]
#dmcdindia1

**Comment by milindsingh** [comment]
@engcom-backlog-nazar @orlangur @hostep 

Magento Admin Shipment form does not allow the shipment to be created  **without Items or Zero qty** items,
But the same can be done via API, causing an error in Admin.

Should we validate shipment create via API? (As I am already working on it, I need to know if it's done intentionally )

Thanks :)

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for your report. 
The issue has been fixed in magento/magento2#22687 by @milindsingh in 2.3-develop branch
Related commit(s):
 - [2e344f1f8bf26571bc0d41e65b4a34c6d46273fc](https://github.com/magento/magento2/commit/2e344f1f8bf26571bc0d41e65b4a34c6d46273fc)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for your report. 
The issue has been fixed in magento/magento2#23119 by @milindsingh in 2.2-develop branch
Related commit(s):
 - [044befab117b5095e04c8dd086c9f2dd4784e573](https://github.com/magento/magento2/commit/044befab117b5095e04c8dd086c9f2dd4784e573)

The fix will be available with the upcoming 2.2.10 release.
